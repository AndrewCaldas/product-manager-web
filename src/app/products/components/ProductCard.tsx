import { Product } from '@/types/product';

interface Props {
  product: Product;
}

export function ProductCard({ product }: Props) {
  const { name, category, description, price, imageUrl } = product;
  return (
    <div className="border rounded-xl p-4 shadow-sm space-y-2">
      <img src={imageUrl} alt={name} className="w-full h-40 object-cover rounded" />
      <h2 className="text-lg font-semibold">{name}</h2>
      <p className="text-sm text-gray-500">{category}</p>
      <p className="text-sm">{description}</p>
      <p className="font-bold">R$ {price.toFixed(2)}</p>
    </div>
  );
}
