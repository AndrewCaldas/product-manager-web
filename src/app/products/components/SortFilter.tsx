'use client';

import { useProductStore } from '@/store/productStore';
import {
  Select,
  SelectTrigger,
  SelectValue,
  SelectContent,
  SelectItem,
} from '@/components/ui/select';

export function SortFilter() {
  const sort = useProductStore((state) => state.sort);
  const setSort = useProductStore((state) => state.setSort);

  return (
    <div className="w-full max-w-xs">
      <Select value={sort} onValueChange={setSort}>
        <SelectTrigger className="w-full">
          <SelectValue placeholder="Ordenar por" />
        </SelectTrigger>
        <SelectContent>
          <SelectItem value="name-asc">Nome (A-Z)</SelectItem>
          <SelectItem value="name-desc">Nome (Z-A)</SelectItem>
          <SelectItem value="price-asc">Preço (menor ao maior)</SelectItem>
          <SelectItem value="price-desc">Preço (maior ao menor)</SelectItem>
        </SelectContent>
      </Select>
    </div>
  );
}
