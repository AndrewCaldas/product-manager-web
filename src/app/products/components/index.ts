export * from './ProductCard';
export * from './FilterBar';
export * from './SortFilter';
