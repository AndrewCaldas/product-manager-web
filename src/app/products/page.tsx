'use client';

import { useEffect } from 'react';
import { useProductStore } from '@/store/productStore';
import { ProductCard, FilterBar } from './components';

export default function ProductListPage() {
  const { filteredProducts, loadProducts } = useProductStore();

  useEffect(() => {
    loadProducts();
  }, [loadProducts]);

  return (
    <div className="p-6 space-y-6">
      <h1 className="text-2xl font-bold">Produtos</h1>

      <FilterBar />

      <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-6">
        {filteredProducts?.map((product) => <ProductCard key={product.id} product={product} />)}
      </div>
    </div>
  );
}
