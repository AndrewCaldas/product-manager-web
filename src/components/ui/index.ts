export * from './input';
export * from './label';
export * from './select';
export * from './slider';
export * from './form';
export * from './textarea';
export * from './button';
