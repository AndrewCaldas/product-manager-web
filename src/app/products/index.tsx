'use client';

import { useEffect } from 'react';
import { useProductStore } from '@/store/productStore';
import { fetchMockProducts } from '@/lib/api';
import { ProductCard } from './components/ProductCard';

export default function ProductListPage() {
  const { filteredProducts, loadProducts } = useProductStore();

  useEffect(() => {
    fetchMockProducts().then(loadProducts);
  }, [loadProducts]);

  return (
    <div className="p-6 space-y-6">
      <h1 className="text-2xl font-bold">Produtos</h1>

      <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-6">
        {filteredProducts.map((product) => (
          <ProductCard key={product.id} product={product} />
        ))}
      </div>
    </div>
  );
}
