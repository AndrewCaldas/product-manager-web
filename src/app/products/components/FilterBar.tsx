'use client';

import { useState, useEffect } from 'react';
import { Input, Slider, Label } from '@/components/ui';
import { useProductStore } from '@/store/productStore';

export function FilterBar() {
  const { search, priceRange, setSearch, setPriceRange } = useProductStore();

  const [localPrice, setLocalPrice] = useState<[number, number]>(priceRange);

  useEffect(() => {
    const debounce = setTimeout(() => {
      setPriceRange(localPrice);
    }, 400);

    return () => clearTimeout(debounce);
  }, [localPrice]);

  return (
    <div className="flex flex-col md:flex-row gap-4 items-center py-4">
      <Input
        placeholder="Buscar por nome"
        value={search}
        onChange={(e) => setSearch(e.target.value)}
        className="max-w-sm"
      />

      <div className="flex flex-col gap-1 w-full max-w-sm">
        <Label>
          Faixa de pre√ßo: R$ {localPrice[0]} - R$ {localPrice[1]}
        </Label>
        <Slider
          min={0}
          max={5000}
          step={50}
          value={localPrice}
          onValueChange={(val) => setLocalPrice(val as [number, number])}
        />
      </div>
    </div>
  );
}
