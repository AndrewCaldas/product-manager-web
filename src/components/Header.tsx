'use client';

import Link from 'next/link';
import { usePathname } from 'next/navigation';
import { Button } from '@/components/ui';

export function Header() {
  const pathname = usePathname();

  return (
    <header className="bg-white shadow-sm py-4 px-6 mb-6">
      <div className="max-w-6xl mx-auto flex justify-between items-center">
        <Link href="/products" className="text-xl font-bold">
          Gerenciador de Produtos
        </Link>

        {pathname === '/products' && (
          <Link href="/products/new">
            <Button variant="default">Adicionar Produto</Button>
          </Link>
        )}

        {pathname === '/products/new' && (
          <Link href="/products">
            <Button variant="outline">Voltar para lista</Button>
          </Link>
        )}
      </div>
    </header>
  );
}
