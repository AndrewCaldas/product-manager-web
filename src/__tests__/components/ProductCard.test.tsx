import { render } from '@testing-library/react';
import { ProductCard } from '@/app/products/components';
import { Product } from '@/types';

let mockProduct: Product;

beforeEach(() => {
  mockProduct = {
    id: '1',
    name: 'Notebook Gamer',
    category: 'EletrÃ´nicos',
    price: 4500,
    description: 'Um notebook potente para jogos.',
    imageUrl: 'https://picsum.photos/300/200',
  };
});

describe('ProductCard', () => {
  it('deve renderizar corretamente e bater o snapshot', () => {
    const { container } = render(<ProductCard product={mockProduct} />);
    expect(container).toMatchSnapshot();
  });
});
