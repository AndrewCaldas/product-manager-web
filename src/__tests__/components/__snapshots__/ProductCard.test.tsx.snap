// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard deve renderizar corretamente e bater o snapshot 1`] = `
<div>
  <div
    class="border rounded-xl p-4 shadow-sm space-y-2"
  >
    <img
      alt="Notebook Gamer"
      class="w-full h-40 object-cover rounded"
      src="https://picsum.photos/300/200"
    />
    <h2
      class="text-lg font-semibold"
    >
      Notebook Gamer
    </h2>
    <p
      class="text-sm text-gray-500"
    >
      Eletr√¥nicos
    </p>
    <p
      class="text-sm"
    >
      Um notebook potente para jogos.
    </p>
    <p
      class="font-bold"
    >
      R$ 
      4500.00
    </p>
  </div>
</div>
`;
